File: ambeth/Ambeth.Merge/ambeth/metadata/IObjRefFactory.cs
using De.Osthus.Ambeth.Cache;
using De.Osthus.Ambeth.Merge.Model;
using System;

namespace De.Osthus.Ambeth.Metadata
{
    public abstract class IObjRefFactory
    {
        public abstract IPreparedObjRefFactory PrepareObjRefFactory(Type entityType, int idIndex);

        public abstract IObjRef CreateObjRef(Type entityType, int idIndex, Object id, Object version);

        public abstract IObjRef CreateObjRef(AbstractCacheValue cacheValue);

        public abstract IObjRef Dup(IObjRef objRef);
    }
}

File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/metadata/IObjRefFactory.java
package de.osthus.ambeth.metadata;

import de.osthus.ambeth.cache.AbstractCacheValue;
import de.osthus.ambeth.merge.model.IObjRef;

public abstract class IObjRefFactory
{
	public abstract IPreparedObjRefFactory prepareObjRefFactory(Class<?> entityType, int idIndex);

	public abstract IObjRef createObjRef(Class<?> entityType, int idIndex, Object id, Object version);

	public abstract IObjRef createObjRef(AbstractCacheValue cacheValue);

	public abstract IObjRef createObjRef(AbstractCacheValue cacheValue, int idIndex);

	public abstract IObjRef dup(IObjRef objRef);
}
